tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientStreamTest.java

```diff
@@ -680,20 +678,29 @@ public void handle(String target, org.eclipse.jetty.server.Request baseRequest,
             }
         });
 
-        CountDownLatch latch = new CountDownLatch(1);
-        try (AsyncRequestContent content = new AsyncRequestContent())
+        CountDownLatch demandLatch = new CountDownLatch(2);
+        CountDownLatch requestLatch = new CountDownLatch(1);
+        try (AsyncRequestContent content = new AsyncRequestContent()
+        {
+            @Override
+            public void demand()
+            {
+                demandLatch.countDown();
+                super.demand();
+            }
+        })
         {
             scenario.client.newRequest(scenario.newURI())
                 .scheme(scenario.getScheme())
                 .body(content)
                 .send(result ->
                 {
                     if (result.isSucceeded() && result.getResponse().getStatus() == 200)
-                        latch.countDown();
+                        requestLatch.countDown();
                 });
 
             // Make sure we provide the content *after* the request has been "sent".
-            Thread.sleep(1000);
+            assertTrue(demandLatch.await(5, TimeUnit.SECONDS));
```

Comment:
Again, the latch could be counted down really quickly, so the content will be available before the request is sent, contrary to the purpose of the test.

Reply:
Indeed. Counting down the latch in a `onRequestCommit` listener would solve that problem, wouldn't it?